velox/common/base/tests/FsTest.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/Fs.h"
+#include <gtest/gtest.h>
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+
+namespace facebook::velox::common {
+
+class FsTest : public testing::Test {
+};
+
+TEST_F(FsTest, createDirectory) {
+  auto tmpDirectoryPath = exec::test::TempDirectoryPath::createTempDirectory();
+  // First time should generate directory successfully.
+  EXPECT_TRUE(generateFileDirectory(tmpDirectoryPath.c_str()));
```

Comment:
I believe createTempDirectory actually creates a directory. If that's the case, this test doesn't verify that generateFileDirectory creates a directory if one doesn't exist. How about creating a a/b/d sub-directory in tmpDirectoryPath?